### PR TITLE
Some more base fees.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by chriseth** [comment]
Not important for the release.

**Review by leonardoalt — DISMISSED** [review]
Nice!

**Comment by leonardoalt** [comment]
Tests need update

**Comment by leonardoalt** [comment]
I'll fix the tests

**Review by leonardoalt — APPROVED** [review]

**Comment by chriseth** [comment]
@leonardoalt can you explain why there are so many test updates needed in tests that do not even use `basefee`?

File: test/cmdlineTests/standard_model_checker_divModSlacks_default_all/output.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
 (declare-datatypes ((|state_type| 0)) (((|state_type| (|balances| (Array Int Int))))))
 (declare-fun |state_0| () |state_type|)
 (declare-datatypes ((|bytes_tuple| 0)) (((|bytes_tuple| (|bytes_tuple_accessor_array| (Array Int Int)) (|bytes_tuple_accessor_length| Int)))))
-(declare-datatypes ((|tx_type| 0)) (((|tx_type| (|block.chainid| Int) (|block.coinbase| Int) (|block.difficulty| Int) (|block.gaslimit| Int) (|block.number| Int) (|block.timestamp| Int) (|blockhash| (Array Int Int)) (|msg.data| |bytes_tuple|) (|msg.sender| Int) (|msg.sig| Int) (|msg.value| Int) (|tx.gasprice| Int) (|tx.origin| Int)))))
+(declare-datatypes ((|tx_type| 0)) (((|tx_type| (|block.basefee| Int) (|block.chainid| Int) (|block.coinbase| Int) (|block.difficulty| Int) (|block.gaslimit| Int) (|block.number| Int) (|block.timestamp| Int) (|blockhash| (Array Int Int)) (|msg.data| |bytes_tuple|) (|msg.sender| Int) (|msg.sig| Int) (|msg.value| Int) (|tx.gasprice| Int) (|tx.origin| Int)))))
```
Comment (leonardoalt) [inline test/cmdlineTests/standard_model_checker_divModSlacks_default_all/output.json:12]: @chriseth because of this

Comment (leonardoalt) [reply test/cmdlineTests/standard_model_checker_divModSlacks_default_all/output.json:12]: you added `basefee` as a transaction variable in the SMT encoding, this effectively changes every single SMT query. When the query changes syntactically, 1) the result proved/unproved might change 2) the counterexamples might change

Comment (chriseth) [reply test/cmdlineTests/standard_model_checker_divModSlacks_default_all/output.json:12]: ah ok!

**Pull request merged**